include/irr/core/IReferenceCounted.h

```diff
@@ -266,11 +266,19 @@ namespace core
 			inline bool operator>(const smart_refctd_ptr<U>& other) const { return ptr > other.ptr; }
 	};
 
+    // create and object and make a smart pointer without increasing reference count above 1

```

Comment:
why did the fix get reverted?

Reply:
Commit message ;) crashes on fence-related code deep in the engine and can't figure why

Reply:
because the fence stuff worked with a fucked up reference count ;)

Reply:
shit, we're leaking fences